plugins/sql_functions.cc

```diff
@@ -611,7 +614,7 @@ run(const ProcedureRunConfig & run,
                          *runProcConf.inputData.stm->where,
                          runProcConf.inputData.stm->orderBy,
                          { runProcConf.inputData.stm->rowName })
-            .execute({recordRowInOutputDataset, true /*processInParallel*/},
+            .execute({recordRowInOutputDataset, !runProcConf.deterministicOrdering /*processInParallel*/},
```

Comment:
It was already executed in parallel.  The flag will only allow it to go slower.  That does not seem to fit the JIRA description!